File: starter_kits/Go/src/hlt/Command.go
package hlt

import "fmt"

// Command - Interface for generating text commands to print out
type Command interface {
	CommandString() string
}

// SpawnShip - Command to spawn a ship
type SpawnShip struct {
}

// CommandString - Returns the text string for the TransformToDropoff command
func (t SpawnShip) CommandString() string {
	return fmt.Sprintf("g")
}

// TransformToDropoff - Command to transform ship into a dropoff point
type TransformToDropoff struct {
	id int
}

// CommandString - Returns the text string for the TransformToDropoff command
func (t TransformToDropoff) CommandString() string {
	return fmt.Sprintf("c %d", t.id)
}

// Move - A command that moves an entity a direction
type Move struct {
	id        int
	direction *Direction
}

// CommandString - Returns the text string for the Move command
func (m Move) CommandString() string {
	return fmt.Sprintf("m %d %c", m.id, m.direction.charValue)
}

File: starter_kits/C#/Halite-3/hlt/Command.cs
using System;
using System.Collections.Generic;
using System.Collections;
using System.Text;

namespace MyBot.hlt
{
    class Command
    {
        public string command;

        public static Command spawnShip()
        {
            return new Command("g");
        }

        public static Command transformShipIntoDropoffSite(EntityId id)
        {
            return new Command("c " + id);
        }

        public static Command move(EntityId id, Direction direction)
        {
            return new Command("m " + id + ' ' + direction.charValue);
        }

        private Command(string command)
        {
            this.command = command;
        }

        public override bool Equals(Object o)
        {
            if (this == o) return true;
            //if (o == null || getClass() != o.getClass()) return false;

            Command command1 = (Command)o;

            return command.Equals(command1.command);
        }

        public override int GetHashCode()
        {
            return command.GetHashCode();
        }
    }
}
